src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorsShouldHaveSymmetricalOverloads.Fixer.cs

```diff
@@ -16,15 +24,96 @@ public abstract class OperatorsShouldHaveSymmetricalOverloadsFixer : CodeFixProv
+                    c => CreateChangedDocument(context, c),
+                    nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.Generate_missing_operators)),
+                context.Diagnostics);
+            return Task.FromResult(true);
+        }
+
+        private async Task<Document> CreateChangedDocument(
+            CodeFixContext context, CancellationToken cancellationToken)
+        {
+            var document = context.Document;
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+            var semanticModel = editor.SemanticModel;
+            var root = await semanticModel.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var operatorNode = root.FindNode(context.Diagnostics.First().Location.SourceSpan);
+
+            var containingOperator = (IMethodSymbol)semanticModel.GetDeclaredSymbol(operatorNode, cancellationToken);
+
+            Debug.Assert(containingOperator.IsUserDefinedOperator());
+
+            var generator = editor.Generator;
+            var newOperator = generator.OperatorDeclaration(
+                GetInvertedOperator(containingOperator),
+                containingOperator.GetParameters().Select(p => generator.ParameterDeclaration(p)),
+                generator.TypeExpression(containingOperator.ReturnType),
+                containingOperator.DeclaredAccessibility,
+                generator.GetModifiers(operatorNode),
+                GetInvertedStatements(generator, containingOperator, semanticModel.Compilation));
+
+            operatorNode = operatorNode.AncestorsAndSelf().First(a => a.RawKind == newOperator.RawKind);
+
+            editor.InsertAfter(operatorNode, newOperator);
+            return editor.GetChangedDocument();
+        }
+
+        private IEnumerable<SyntaxNode> GetInvertedStatements(
+            SyntaxGenerator generator, IMethodSymbol containingOperator, Compilation compilation)
+        {
+            yield return GetInvertedStatement(generator, containingOperator, compilation);
+        }
+
+        private SyntaxNode GetInvertedStatement(
+            SyntaxGenerator generator, IMethodSymbol containingOperator, Compilation compilation)
+        {
+            if (containingOperator.Name == WellKnownMemberNames.EqualityOperatorName)
+            {
+                return generator.ReturnStatement(
+                    generator.LogicalNotExpression(
+                        generator.ValueEqualsExpression(
+                            generator.IdentifierName(containingOperator.Parameters[0].Name),
+                            generator.IdentifierName(containingOperator.Parameters[1].Name))));
+            }
+            else if (containingOperator.Name == WellKnownMemberNames.InequalityOperatorName)
+            {
+                return generator.ReturnStatement(
+                    generator.LogicalNotExpression(
```

Comment:
Why the double negation? Can't it be just,

```C#
    if (containingOperator.Name == WellKnownMemberNames.EqualityOperatorName)
    {
        return generator.ReturnStatement(
            generator.ValueNotEqualsExpression(
                generator.IdentifierName(containingOperator.Parameters[0].Name),
                generator.IdentifierName(containingOperator.Parameters[1].Name)));
    }
```

similarly for the other condition